### PR TITLE
Add rsc and authorize query call

### PR DESCRIPTION
**Pull request by MMarelli**
- Adds RSC and configuration
- Adds call to authorize query for rewrite

For authorize query the call is done as the first step when loading the table. The rewritten query is then sent to BQ with the result going to a temp table
The resulting DF is the temp table that can be used without further auth as all okera policies have been applied

### CODE REVIEW

File: spark-bigquery-connector-common/pom.xml [hunk 1]
```diff
@@ -18,6 +18,25 @@
         </license>
     </licenses>
     <dependencies>
+        <!-- Start: Okera Dependencies -->
+        <dependency>
+            <groupId>com.okera.recordservice</groupId>
+            <artifactId>recordservice-core</artifactId>
+            <version>2.11.0-SNAPSHOT</version>
```
Comment (MMarelli) [inline spark-bigquery-connector-common/pom.xml:25]: Need new RSC published for this to build

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/okera/OptionalUtil.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.google.cloud.spark.bigquery.okera;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class OptionalUtil {
+  public static <T> Optional<? extends T> or(
```
Comment (csunwold) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/okera/OptionalUtil.java:7]: Isn't this basically the same as orElseGet? https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseGet-java.util.function.Supplier-

Comment (MMarelli) [reply spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/okera/OptionalUtil.java:7]: It's close not the same. It maintains the optional type so the fallback can be optional as well. It's actually part of optional in 11 https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#or(java.util.function.Supplier)

File: pom.xml [hunk 3]
```diff
@@ -139,4 +139,53 @@
...
+      <name>libs-release</name>
+      <url>https://cerebro.jfrog.io/cerebro/libs-release</url>
+    </repository>
+    <repository>
+      <id>cerebro.snapshots.repo</id>
+      <name>libs-snapshot</name>
+      <url>https://cerebro.jfrog.io/cerebro/libs-snapshot</url>
+    </repository>
+    <repository>
+      <id>cdh.rcs.releases.repo</id>
```
Comment (saurabh279) [inline pom.xml:155]: we can remove some of these. don't think spark bq connector will have cdh deps

**Pull request merged**